src/Shared/WebEncoders/WebEncoders.cs

```diff
@@ -0,0 +1,388 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.Extensions.WebEncoders.Sources;
+
+#if WebEncoders_In_WebUtilities
+namespace Microsoft.AspNetCore.WebUtilities
+#else
+namespace Microsoft.Extensions.Internal
+#endif
+{
+    /// <summary>
+    /// Contains utility APIs to assist with common encoding and decoding operations.
+    /// </summary>
+#if WebEncoders_In_WebUtilities
+    public
+#else
+    internal
+#endif
+    static class WebEncoders
```

Comment:
In the ongoing aspnet<->dotnet repository refactoring, would it make sense to move base64 encode/decode URL string APIs to aspnet/extensions (eventually dotnet/runtime)? Many non-web,mvc applications (microservice, management tools, data processing etc.) have to deal with such encodings. IMO, it would be great if the entire RFC 4648 implementation comes from one place (corefx libraries), rather than scattered across two worlds.